tests/sentry/api/endpoints/test_organization_index.py

```diff
@@ -105,6 +123,34 @@ def test_without_slug(self):
         org = Organization.objects.get(id=organization_id)
         assert org.slug == "hello-world"
 
+    def test_name_slugify(self):
+        response = self.get_success_response(name="---foo")
+        org = Organization.objects.get(id=response.data["id"])
+        assert org.slug == "foo"
+
+        org_slug_pattern = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-]*(?<!-)$")
+
+        response = self.get_success_response(name="---foo---")
+        org = Organization.objects.get(id=response.data["id"])
+        assert org.slug != "foo-"
+        assert org.slug.startswith("foo-")
+        assert org_slug_pattern.match(org.slug)
+
+        response = self.get_success_response(name="___foo___")
+        org = Organization.objects.get(id=response.data["id"])
+        assert org.slug != "foo-"
+        assert org.slug.startswith("foo-")
+        assert org_slug_pattern.match(org.slug)
+
+        response = self.get_success_response(name="foo_bar")
+        org = Organization.objects.get(id=response.data["id"])
+        assert org.slug == "foo-bar"
+
+        response = self.get_success_response(name="----")
+        org = Organization.objects.get(id=response.data["id"])
+        assert len(org.slug) > 0
```

Comment:
@markstory There cannot be an assertion of what the generated slug will be specifically since it's a randomized string.

Here, the original org slug, `----`, will become a blank string, and that blank string will be sent to `slugify_instance()`, and then it gets turned into a randomized string of 11 characters here: https://github.com/getsentry/sentry/blob/0a4a14b9eecb82e2ad990c2bd551ee81e5b53416/src/sentry/db/models/utils.py#L66-L67

The next line should assert that the final org slug is legal.